packages/webview_flutter/webview_flutter/lib/src/v4/src/webview_controller.dart

```diff
@@ -0,0 +1,263 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+// TODO(a14n): remove this import once Flutter 3.1 or later reaches stable (including flutter/flutter#104231)
+// ignore: unnecessary_import
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:webview_flutter_platform_interface/v4/webview_flutter_platform_interface.dart';
+
+/// Controls a WebView provided by the host platform.
+///
+/// Pass this to a [WebViewWidget] to display the WebView.
+class WebViewController {
+  /// Constructs a [WebViewController].
+  WebViewController()
+      : this.fromPlatform(
+          platform: PlatformWebViewController(
+            const PlatformWebViewControllerCreationParams(),
+          ),
+        );
+
+  /// Constructs a [WebViewController] from creation params for a specific
+  /// platform.
+  WebViewController.fromPlatformCreationParams(
+    PlatformWebViewControllerCreationParams params,
+  ) : this.fromPlatform(platform: PlatformWebViewController(params));
+
+  /// Constructs a [WebViewController] from a specific platform implementation.
+  WebViewController.fromPlatform({required this.platform});
+
+  /// Implementation of [PlatformWebViewController] for the current platform.
+  final PlatformWebViewController platform;
```

Comment:
Why is this public? I may just be misremembering the architecture, but I thought we didn't want to expose the platform interface layer APIs.

Reply:
Oh, I remember; we decided we were going to try letting people use the underlying platform object for extensions instead of using a separate extension object, and compare that with the extension object model in actual use.